www/js/crypter/crypto_app.js

```diff
@@ -51,11 +53,45 @@ window.filesender.crypto_app = function () {
             v2019_generated_password_that_is_full_256bit: 2
         },
 
+        createChunkIDArray: function( chunkid ) {
+            var ret = new Uint8Array(4);
+            ret[0] = chunkid>>0  & 0xFF;
+            ret[1] = chunkid>>8  & 0xFF;
+            ret[2] = chunkid>>16 & 0xFF;
+            ret[3] = chunkid>>24 & 0xFF;
+            return ret;
+        },
+
+        /**
+         * Create and return an IV of 16 octets (128 bits) constructed as follows:
+         *    12 octets of entropy
+         *     4 octets containing the chunkid
+         *
+         * This is based on Page 19 of OpenFortress 2018 document:
+         *   "The suggested procedure for the case of FileSender is to combine 96 bits
+         *    of random material with a 32-bit chunk counter to form a 128-bit IV."
+         */
```

Comment:
Please also note this remark on the bottom of Page 19:

  * The 96 bits worth of random material in the IV are produced from a secure random generator, and kept constant for one file (or one upload, as desired). This 96-bit value can be used to see if chunks belong to the same file or upload.

If I understand correctly, you are generating random material for each chunk independently.  As a result, you cannot checking the 96 bits to match upon recollection of chunks.  This means that chunks might be taken out of one upload and substituted in another upload.